### PR TITLE
#168315879 Implement Reset Password Page

### PR DESCRIPTION
**Pull request by ayodejiAA**
#### What does this PR do?
- Implement reset password page
- Add unit tests

#### Description of Task proposed in this pull request?
- Add actions and reducers
- Implement the reset password page
- Add unit tests for actions, reducers and components

#### How should this be manually tested (Quality Assurance)?
- Run `npm run start:dev'
- A browser tab will launch, then navigate to this route ```/request-password``` to request for a password reset link
- Input a valid email address and confirm the email address is a registered one, then submit.
- If successful, check your mail to get the link.
- Copy the token on the url sent and add to this route- ```/reset-password?token=< token here>```, open the route and change to a new password

#### What are the relevant pivotal tracker stories?
- [#168315879](https://www.pivotaltracker.com/story/show/168315879)

#### Any background context you want to add (Operations Impact)?
- Run ```npm install``` to install all necessary dependencies.

#### What I have learned working on this feature: 
- I understood how redux works and state management

#### Screenshots: [If you made some visual changes to the application please upload screenshots here, or remove this section]
Screenshot 2019-09-08 at 7 12 00 PM

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -20,11 +20,12 @@
     "test": {
       "plugins": [
         ["@babel/plugin-proposal-class-properties", { "loose": true }],
-        "@babel/plugin-transform-runtime",
+        "@babel/plugin-transform-runtime"
       ]
     },
     "production": {
       "plugins": [
+        "@babel/plugin-proposal-class-properties",
```
Comment (abdulfatai360) [inline .babelrc:28]: @ayodejiAA Do we need this plugin in production? I believe our app didn't break on Heroku when we didn't add the plugin. Could you clarify, please?

File: src/components/Input/Input.jsx [hunk 2]
```diff
@@ -33,17 +33,20 @@ Input.defaultProps = {
...
+  placeholder: '',
 };
 
+
 Input.propTypes = {
   type: PropTypes.string.isRequired,
   value: PropTypes.string.isRequired,
   name: PropTypes.string.isRequired,
-  placeholder: PropTypes.string.isRequired,
+  placeholder: PropTypes.string,
```
Comment (abdulfatai360) [inline src/components/Input/Input.jsx]: @ayodejiAA Please, why did you remove the placeholder attribute as a required prop? I feel placeholder should be a  required prop because since our form inputs don't have labels, we need a way to give users a clue of the type of information they are required to fill in the input.

File: .babelrc (hunk 1 above)
Comment (ayodejiAA) [reply .babelrc:28]: Build failed when it was not added. It is necessary we have it so we can use state in components without having a constructor.

File: src/styles/variables.scss [hunk 3]
```diff
@@ -1,5 +1,5 @@
-
 $mustardColor: #D7B914;
 $greyColor: #C4C4C4;
 $matterhornColor: #505050;
 $whiteColor:#fff;
+$redColor: #dd2c00
```
Comment (abdulfatai360) [inline src/styles/variables.scss]: Could you add a new line here, please?

File: .babelrc (hunk 1 above)
Comment (abdulfatai360) [reply .babelrc:28]: Alright then.

File: src/styles/variables.scss (hunk 3 above)
Comment (ayodejiAA) [reply src/styles/variables.scss]: Done.

**Review by abdulfatai360 — APPROVED** [review]

**Review by henryade — APPROVED** [review]

File: src/components/ResetPassword/ResetPassword.jsx [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+  response: {}
+};
+
+/**
+ * @returns {function} mount component
+ *
+ * @memberof ResetPassword
+ */
+componentDidMount = () => {
+  const { props: { location: { search } } } = this;
```
Comment (dinobi) [inline src/components/ResetPassword/ResetPassword.jsx:36]: I really don't understand this `search`

File: src/components/ResetPassword/ResetPassword.scss [hunk 5]
```diff
@@ -0,0 +1,37 @@
+.form-wrapper {
+	display: flex;
+	justify-content: center;
+	align-items: center;
+	flex-direction: column;
+}
+
```
Comment (dinobi) [inline src/components/ResetPassword/ResetPassword.scss:7]: I think the spaces between style definitions is much

File: src/reducers/user/resetPassword.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+  RESET_PASSWORD_SUCCESS
+} from '../../actionTypes/index';
+
+const initialState = { isLoading: false };
+
+export default (state = initialState, action) => {
+  switch (action.type) {
+  case RESET_PASSWORD_REQUEST:
+    return {
+      isLoading: true
```
Comment (dinobi) [inline src/reducers/user/resetPassword.js]: you are mutating state here, I think this should rather be:
```
return { 
   ...state,
   isLoading: true,
};
```

File: src/components/ResetPassword/ResetPassword.jsx (hunk 4 above)
Comment (ayodejiAA) [reply src/components/ResetPassword/ResetPassword.jsx:36]: Because I append the reset password token to the url with a query string, search in location object is referencing the token.

File: src/reducers/user/resetPassword.js (hunk 6 above)
Comment (ayodejiAA) [reply src/reducers/user/resetPassword.js]: Updated @dinobi

File: src/actions/user/resetPassword.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+} from '../../actionTypes/index';
+
+export const resetPassword = () => ({
+  type: RESET_PASSWORD_REQUEST,
+  payload: {}
+});
+
+export const success = (data) => ({
+  type: RESET_PASSWORD_SUCCESS,
+  payload: { ...data }
```
Comment (dinobi) [inline src/actions/user/resetPassword.js]: There's no need spreading here, the success function can be:
```
export const success = (payload) => ({
  type: RESET_PASSWORD_SUCCESS,
  payload
});
```

**Review by Adekoreday — APPROVED** [review]
